server.js

```diff
@@ -1,103 +1,169 @@
 var express = require('express'),
     http = require('http');
-    bodyParser = require('body-parser'),
+bodyParser = require('body-parser'),
     proxy = require('express-http-proxy'),
     urlHelper = require('url');
+const dotenv = require('dotenv');
+dotenv.config();
 const latexService = require('./latexService.js')
 
 // ENV Variables
-const BASE_URL = 'dock.sunbirded.org';
-const API_AUTH_TOKEN = "";
-const PORTAL_COOKIES= ""
-const USER_TOKEN = "";
-
+const BASE_URL = process.env.BASE_URL;
+const API_AUTH_TOKEN = process.env.AUTH_API_TOKEN;
+const USER_TOKEN = process.env.USER_API_TOKEN;
 
 var app = express();
 app.set('port', 3000);
 app.use(express.json())
 app.get("/latex/convert", latexService.convert)
 app.post("/latex/convert", bodyParser.json({ limit: '1mb' }), latexService.convert);
-app.all(['/api/framework/v1/read/*',
-     '/learner/framework/v1/read/*', 
-     '/api/channel/v1/read/*'], proxy(BASE_URL, {
-    https: true,
-    proxyReqPathResolver: function(req) {
-        console.log('proxyReqPathResolver ',  urlHelper.parse(req.url).path);
-        return urlHelper.parse(req.url).path;
-    },
-    proxyReqOptDecorator: function(proxyReqOpts, srcReq) {
-        console.log('proxyReqOptDecorator 2')
-        // you can update headers
-        proxyReqOpts.headers['Content-Type'] = 'application/json';
-        proxyReqOpts.headers['user-id'] = 'content-editor';
-        proxyReqOpts.headers['Cookie'] = PORTAL_COOKIES;
-        proxyReqOpts.headers['authorization'] = `Bearer ${API_AUTH_TOKEN}`;
+app.use(express.static(__dirname + '/dist/sunbird-collection-editor/'));
+
+var publicRequestHeaders = {
+    authorization: `Bearer ${API_AUTH_TOKEN}`,
+    "x-authenticated-user-token": USER_TOKEN,
+    "x-channel-id": '01374279726315929680',
+    "user-id": 'collection-editor'
+};
+
+var contentTypeHeaders = {
+    'content-type': "application/json"
+}
+
+const decoratePublicRequestHeaders = function () {
+    return function (proxyReqOpts, srcReq) {
+        proxyReqOpts.headers = Object.assign({}, proxyReqOpts.headers, publicRequestHeaders, contentTypeHeaders);
         return proxyReqOpts;
     }
-}));
+}
+
+const customDecorateReqHeaders = function () {
+    return function (proxyReqOpts, srcReq) {
+        proxyReqOpts.headers = Object.assign({}, proxyReqOpts.headers, publicRequestHeaders);
+        return proxyReqOpts;
+    }
+}
+
+app.post(['/action/content/v3/upload/url/*'], proxy(BASE_URL, {
```

Comment:
Why did we onboarded these API's now. How it was working before?

Reply:
Before It was not working locally so we have fixed all the APIs in order to work locally.